dotnet/FieldEncryptionRsa.cs

```diff
@@ -0,0 +1,133 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Security.Cryptography.X509Certificates;
+using System.Text;
+using System.Threading.Tasks;
+using Couchbase.Configuration.Client;
+using Couchbase.Extensions.Encryption;
+using Couchbase.Extensions.Encryption.Providers;
+using Couchbase.Extensions.Encryption.Stores;
+using Newtonsoft.Json;
+
+namespace DevGuide
+{
+    /*********************************PLEASE READ!***********************************
+     * In order to run this example, you will need to generate an X509 certificate in
+     * the .pfx (Personal Information Exchange Format) and then change it's 'Copy to
+     * to Output Directory' flag to 'Copy Always' in the Properties Window in VS. To do
+     * this make sure you have OpenSsl installed: https://wiki.openssl.org/index.php/Binaries.
+     * Then open open a command prompt and do the following steps:
+     *
+     * Step 1 - Generate an RSA private key:
+     *
+     *          openssl genrsa -out private.key 2048
+     *
+     * Step 2 - Optionally remove the password:
+     *
+     *          openssl rsa -in private.key -out private-nokey.pem
+     *
+     * Step 3 - Generate a new x509 certificate from the private key.  You will have to enter quite a bit of
+     *          information (country, state, company, etc):
+     *
+     *          openssl req -new -x509 -key private.key -out publickey.cer -days 365
+     *
+     * Step 4 - Export the private key and the x509 certificate into a .pfx using 'password' as the password:
+     *
+     *          openssl pkcs12 -export -out public_privatekey.pfx -inkey private.key -in publickey.cer
+     *
+     * Once you have done step 4, copy the public_privatekey.pfx into the DevGuide project and change its "Copy
+     * To Output Directory" flag to "Copy Always".
+     */
```

Comment:
@jeffrymorris, could you write a script and put it into `/etc` directory of this repository, so every project could use the same setup. And users won't need to go into source code to extract deployment steps.

Reply:
@avsej yes, I will do that.